### ISSUE TITLE
ResourceRequest timed out on Failed auth

### ISSUE THREAD

**Issue by imack** [jbielick/faktory_worker_node#353]
node.js version: 14.17.1
npm/yarn version: 7.6.1
faktory-server version: 1.6.1
`facktory-worker` package version:4.5.0"

I was trying to connect via a `FAKTORY_URL` to a staging environment and forgot to put the basic auth in the url. However, the error I got back was: 

```
[2022-10-03 20:59:27.458] [ERROR] ResourceRequest timed out | {"name":"TimeoutError","stack":"TimeoutError: ResourceRequest timed out\n    at ResourceRequest._fireTimeout (<project root>/node_modules/faktory-worker/node_modules/generic-pool/lib/ResourceRequest.js:62:17)\n    at Timeout.bound <project root>/node_modules/faktory-worker/node_modules/generic-pool/lib/ResourceRequest.js:8:15)\n    at listOnTimeout (internal/timers.js:557:17)\n    at processTimers (internal/timers.js:500:7)"}
```

ie. a timeout.

It took me a while to figure out the issue. I couldn't see anything in the connection code that would expose this error other than a timeout. Does Faktory not return any kind of auth error to the workers trying to do a `BEAT`?

**Comment by jbielick** [comment]
Thanks for the issue report. It looks like these errors (handshake) weren't surfacing properly, so I've made sure they're logged to stderr in 7a00905. This fix is available in v4.5.1 .

**Comment by imack** [comment]
Thanks @jbielick. Awesome, don't want anyone else to get confused on that one like I did.
